client/network/src/protocol/sync.rs

```diff
@@ -1519,11 +1519,24 @@ impl<B: BlockT> ChainSync<B> {
 		self.pending_requests.set_all();
 		debug!(target:"sync", "Restarted with {} ({})", self.best_queued_number, self.best_queued_hash);
 		let old_peers = std::mem::take(&mut self.peers);
+
```

Comment:
And could you add some docs on this restart function, on what the return means?

I know that you don't have written it, but this makes it easier in the future to understand it.